swift/IceStorm/clock/Sources/Subscriber/main.swift

```diff
@@ -25,150 +28,136 @@ enum Option: String {
     case oneway = "--oneway"
 }
 
-func run(ctrlCHandler: CtrlCHandler) async -> Int32 {
-    do {
-        var args = [String](CommandLine.arguments.dropFirst())
-        let communicator = try Ice.initialize(args: &args, configFile: "config.sub")
-        defer {
-            communicator.destroy()
-        }
+// CtrlCHandler is a helper class that handles Ctrl+C and similar signals. It must be constructed at the beginning
+// of the program, before creating an Ice communicator or starting any thread.
+let ctrlCHandler = CtrlCHandler()
 
-        args = try communicator.getProperties().parseCommandLineOptions(prefix: "Clock", options: args)
-
-        var topicName = "time"
-        var option: Option = .none
-        var batch = false
-        var id: String?
-        var retryCount: String?
-
-        for var i in 0 ..< args.count {
-            let oldoption = option
-            if let o = Option(rawValue: args[i]) {
-                option = o
-            } else if args[i] == "--batch" {
-                batch = true
-            } else if args[i] == "--id" {
-                i += 1
-                if i >= args.count {
-                    usage()
-                    return 1
-                }
-                id = args[i]
-            } else if args[i] == "--retryCount" {
-                i += 1
-                if i >= args.count {
-                    usage()
-                    return 1
-                }
-                retryCount = args[i]
-            } else if args[i].starts(with: "--") {
-                usage()
-                return 1
-            } else {
-                topicName = args[i]
-                i += 1
-                break
-            }
-
-            if oldoption != option, oldoption != .none {
-                usage()
-                return 1
-            }
-        }
+// Create an Ice communicator to initialize the Ice runtime.
+var args = [String](CommandLine.arguments.dropFirst())
```

Comment:
We should show the same in the Greeter demo and hopefully explain this dropFirst.